js/cheap.js

```diff
@@ -0,0 +1,71 @@
+async function donate() {
+  const account = await getAccount();
+  const value = getValue();
+  if (value && account) {
+    await performTransaction(value, account);
+  }
+}
+
+async function getAccount() {
+  // ask which account the user intends to use
+  // TODO: check if Metamask is installed
+  const accounts = await ethereum.request({ method: "eth_requestAccounts" });
+  return accounts[0];
+}
+
+function getValue() {
+  // get the donation value inserted by the user
+  const amount = document.getElementById("donation-input").value;
+  if (isNaN(amount)) {
+    document.getElementById("donation-input").value =
```

Comment:
Something has passed through my mind. Sometimes jquery is picky and to get the value there's (also) another way, which is $('#donation-input').val(). Just to have this in mind ;)

Reply:
Yes indeed I could have used jQuery too given that we have it here :) I have been full vanilla in such cases for a while as it's often been deprecated nowadays (e.g. bootstrap 5)